### PR TITLE
xkbcommon: bump wayland + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

required for https://github.com/conan-io/conan-center-index/pull/8821#issuecomment-1016303191

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xkbcommon/all/conanfile.py [hunk 1]
```diff
@@ -56,23 +56,23 @@ def configure(self):
...
+            raise ConanInvalidConfiguration("This library is only compatible with Linux or FreeBSD")
 
     def build_requirements(self):
-        self.build_requires("meson/0.59.1")
-        self.build_requires("bison/3.7.1")
-        if self.options.get_safe("with_wayland", False):
-            self.build_requires("wayland/1.19.0")
+        self.build_requires("meson/0.60.2")
+        self.build_requires("bison/3.7.6")
+        if hasattr(self, "settings_build") and self.options.get_safe("wayland"):
```
Comment (ericLemanissier) [inline recipes/xkbcommon/all/conanfile.py:74]: why the addition of `hasattr(self, "settings_build")` ?

Comment (SpaceIm) [reply recipes/xkbcommon/all/conanfile.py:74]: it's explained in the last commit

Comment (SpaceIm) [reply recipes/xkbcommon/all/conanfile.py:74]: usual issue when a downstream recipe overrides wayland with 1 profile

Comment (ericLemanissier) [reply recipes/xkbcommon/all/conanfile.py:74]: ok, thanks

**Review by ericLemanissier — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**